openadapt/record.py

```diff
@@ -628,11 +628,44 @@ def record(
 
     # TODO: discard events until everything is ready
 
+    # Variables for sequence detection
+    sequence = ['o', 'a', '.', 's', 't', 'o', 'p']
+    current_index = 0
+    sequence_detected = False  # Flag to indicate if the sequence is detected
+
+    def on_press(key):
+        canonical_key = listener.canonical(key)[0]
+        nonlocal current_index, sequence_detected
+        sequence_canonical = listener.canonical(keyboard.KeyCode.from_char(sequence[current_index]))
```

Comment:
What do you think about using a consistent naming convention here? e.g. `canonical_sequence` to match `canonical_key`